cpp/src/lance/format/schema.cc

```diff
@@ -383,6 +385,38 @@ std::shared_ptr<Field> Field::Project(const std::shared_ptr<::arrow::Field>& arr
   return new_field;
 }
 
+::arrow::Result<std::shared_ptr<Field>> Field::Merge(const ::arrow::Field& arrow_field) const {
+  if (name() != arrow_field.name()) {
+    return ::arrow::Status::Invalid(
+        "Attempt to merge two different fields: ", name(), "!=", arrow_field.name());
+  }
+  auto self_type = type();
+  if (self_type->id() != arrow_field.type()->id()) {
+    return ::arrow::Status::Invalid(
+        "Can not merge two fields: ", self_type->ToString(), "!=", arrow_field.type()->ToString());
```

Comment:
nit: change the message to indicate that it's trying to merge two different **types**

Reply:
done